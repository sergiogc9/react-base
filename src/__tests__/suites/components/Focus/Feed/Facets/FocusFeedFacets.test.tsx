import { shallow } from "enzyme";
import React from "react";
import FocusFeedFacets from "@src/components/Focus/Feed/Facets/FocusFeedFacets";

let onToggleDrawer: any;
describe("<FocusFeedFacets />", () => {
	let wrapper: any;

	beforeAll(() => {
		window.matchMedia = jest.fn((query: string) => ({
			matches: query.indexOf('(min-width: 1025px)') !== -1,
		}));
	});

	beforeEach(() => {
		onToggleDrawer = jest.fn();
		wrapper = shallow(<FocusFeedFacets
			drawerVisible={true}
			onToggleDrawer={onToggleDrawer}
		/>);
	});

	it("Test component snapshot", () => {
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onToggleDrawer).toHaveBeenCalled();
	});

	it("Test component snapshot", () => {
		wrapper.setProps({ drawerVisible: false });
		expect(wrapper.debug()).toMatchSnapshot();
		expect(onToggleDrawer).toHaveBeenCalled();
	});
});
