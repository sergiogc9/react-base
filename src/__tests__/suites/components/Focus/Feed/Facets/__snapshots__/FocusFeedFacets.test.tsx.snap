// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FocusFeedFacets /> Test component snapshot 1`] = `
"<Drawer id=\\"facetsDrawer\\" type=\\"floating\\" inline={true} visible={true} clickableDesktopOverlay={false} overlay={false} onVisibilityChange={[Function: onVisibilityChange]} onMediaTypeChange={[Function: onMediaTypeChange]} defaultMedia=\\"mobile\\" mobileType=\\"temporary\\" mobileMinWidth={320} tabletType=\\"persistent\\" tabletMinWidth={768} desktopType=\\"full-height\\" desktopMinWidth={1025} position=\\"left\\" transitionDuration={300} autoclose={true} constantType={true}>
  <Connect(SearchFacets) />
</Drawer>"
`;

exports[`<FocusFeedFacets /> Test component snapshot 2`] = `
"<Drawer id=\\"facetsDrawer\\" type=\\"floating\\" inline={true} visible={false} clickableDesktopOverlay={false} overlay={false} onVisibilityChange={[Function: onVisibilityChange]} onMediaTypeChange={[Function: onMediaTypeChange]} defaultMedia=\\"mobile\\" mobileType=\\"temporary\\" mobileMinWidth={320} tabletType=\\"persistent\\" tabletMinWidth={768} desktopType=\\"full-height\\" desktopMinWidth={1025} position=\\"left\\" transitionDuration={300} autoclose={true} constantType={true}>
  <Connect(SearchFacets) />
</Drawer>"
`;
